### PR TITLE
XDR-4369: add DCR rule module (feat)

### PR DESCRIPTION
**Pull request by chrisjaimon2012**

### CODE REVIEW

**Review by GerardSetho — DISMISSED** [review]
:shipit:

File: modules/monitor-data-collection-rule/main.tf [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+#  name                    = "${var.name}-association"
+#  description             = var.description
+#  target_resource_id      = var.target_resource_id
+#  data_collection_rule_id = azurerm_monitor_data_collection_rule.dcr.id
+#}
+
+# TODO: add validations for resource_group_name and target_resource_id to prevent potential CLI injection.
+resource "null_resource" "dcr_association" {
+  provisioner "local-exec" {
+    command = "az monitor log-analytics workspace update --resource-group ${var.resource_group_name} --workspace-name ${var.target_resource_id} --data-collection-rule \"${azurerm_monitor_data_collection_rule.dcr.id}\""
```
Comment (GerardSetho) [inline modules/monitor-data-collection-rule/main.tf:242]: Is this to "deploy" the DCR?

I would think applying "azurerm_monitor_data_collection_rule" would deploy the rule to the ALA workspace.

Comment (chrisjaimon2012) [reply modules/monitor-data-collection-rule/main.tf:242]: It doesn't 😢

Comment (chrisjaimon2012) [reply modules/monitor-data-collection-rule/main.tf:242]: Check this out https://github.com/hashicorp/terraform-provider-azurerm/issues/20637

Comment (GerardSetho) [reply modules/monitor-data-collection-rule/main.tf:242]: Can you add some comment for line 242? I'm certain that in 6 months, I'm going to forget and ask myself, why are we doing this? Thanks bro.

Comment (chrisjaimon2012) [reply modules/monitor-data-collection-rule/main.tf:244]: Done 👍🏻

**Review by GerardSetho — APPROVED** [review]
:shipit:

**Pull request merged**